### DISCUSSION TITLE
MissingMethodException: DryIoc.Rules DryIoc.Rules.WithoutFastExpressionCompiler

### DISCUSSION THREAD

**Discussion by schallerdo** [PrismLibrary/Prism#2715] [General] [answered]
Hello,

I have a C# WPF application and want to use the following two NuGet packages:

* "Prism.DryIoc" Version 8.1.97
* "Prism.Wpf"> Version 8.1.97

But when I start the App, I get this exception: MissingMethodException: DryIoc.Rules DryIoc.Rules.WithoutFastExpressionCompiler

What am I doing wrong? Are these packages not to be used together?

Thanks
Dominik

**Comment by dansiegel** [comment, marked as answer]
You've updated DryIoc out of bound. There was a breaking change from 4 to 5. We will be updating the next version of Prism but in the mean time you need to stay on the major version of DryIoc that Prism 8.1 shipped with.

**Reply by schallerdo** [reply]
I don't use DryIoc directly, just your two packages.

**Reply by brianlagunas** [reply]
Please, attach a sample that reproduces the issue.

**Reply by schallerdo** [reply]
I could not reproduce it in a sample. Seems like some other depencency I have used the newer version. Thanks for your help.

**Comment by safarviktor** [comment]
Prism 8.1 comes with DryIoc 4.7.7 (or it did for me)
When I followed https://www.andicode.com/prism/wpf/logging/2021/05/21/Logging-In-Prism.html I blindly installed `DryIoC.Microsoft.DependencyInjection` v6.x which then gave me the OP error.
Downgraded `DryIoC.Microsoft.DependencyInjection` to the latest version of 4 resolved the problem.
